feature-group-definitions/cascade-layers.yml

```diff
@@ -1,2 +1,17 @@
 spec: https://drafts.csswg.org/css-cascade-5/#layering
 caniuse: css-cascade-layers
+constituentFeatures:
+  - source: "@mdn/browser-compat-data"
+    query: css.at-rules.layer
+  - source: "@mdn/browser-compat-data"
+    query: css.at-rules.import.layer
+  - source: "@mdn/browser-compat-data"
+    query: api.CSSImportRule.layerName
+  - source: "@mdn/browser-compat-data"
+    query: api.CSSLayerBlockRule
+  - source: "@mdn/browser-compat-data"
+    query: api.CSSLayerBlockRule.name
+  - source: "@mdn/browser-compat-data"
+    query: api.CSSLayerStatementRule
+  - source: "@mdn/browser-compat-data"
+    query: api.CSSLayerStatementRule.nameList
```

Comment:
I wonder if we can't separate mappings between different data sources and make this one BCD specific for simplicty? Something like:

```
bcdFeatures:
  - css.at-rules.layer
  - css.at-rules.import.layer
  - api.CSSImportRule.layerName
  - api.CSSLayerBlockRule
  - api.CSSLayerBlockRule.name
  - api.CSSLayerStatementRule
  - api.CSSLayerStatementRule.nameList
```

Reply:
I think that's a possibility, though I have a concern about what it implies about the relationship between the feature and its constituents. For instance, if I migrate something here like [the mockup's `array.json`](https://github.com/ddbeck/common-web-feature-mockup/blob/main/web-platform-features/array.json) and we split them up by source, we might end up with something like:

```yaml
localFeatures:
  - array-relative-indexing
  - array-grouping
bcdFeatures:
  - javascript.builtins.Array
  - javascript.builtins.Array.Array
  - javascript.builtins.Array.concat
  - …
```

If we structure the data like this, I'm concerned that it:

- Obfuscates the fact that only `localFeatures` plus `bcdFeatures` describes a complete thing (i.e., neither key alone describes the feature)
- Makes for a bad contribution experience, where it puts the focus on data sources rather than the feature description (put another way: if BCD lacks some data to accurately characterize a feature and its support, that's a problem to fix in BCD, not a constraint on what the feature is)

That said, I recognize that the form I've presented is verbose and repetitious (though I'm not 100% certain if that's your concern, exactly). I think it would be OK to be terser, use short hands, and set defaults (or some combination thereof). Some ideas along these lines:

```yaml
constituents:  # less words
- query: css.at-rules.layer
  # implicit default for `source: "@mdn/browser-compat-data"`?
- bcd:css.at-rules.layer  # use some sort of source:identifier string notation?
- bcd://css.at-rules.layer  # webbier-looking version the previous line?
- css.at-rules.layer  # assume that dotted paths are BCD?

Reply:
@foolip I've switched the name of the new key to underscores and merged in the usage stats from main. I was wondering if you had any more thoughts on the mapping situation. Happy to move forward with any reasonable suggestion (of which they're all reasonable, including separating the data sources). But I wanted to give you another opportunity to chime in before I moved forward on migrating more data out of the mockup.

Reply:
I thought I had replied here last week, but no.

Unless we have another data source in mind that we could use for computing web-feature statuses I think we should simplify and optimize for the case that all constituent features are from BCD.

But I think that if I'm right about this it will quickly become evident just by trying to maintain the data, so I don't need to get my way if your hunch is different.